### PR TITLE
Add Quark Engine as one of the static analyzers

### PR DESCRIPTION
**Pull request by pulorsok**
### Describe the Pull Request

This PR refers to [#1674](https://github.com/MobSF/Mobile-Security-Framework-MobSF/issues/1674)

Hi @ajinabraham, sorry for taking so long to do this.
We (me, @18z, @krnick) have implemented some features for the integration:

#### 1. Add Quark Analysis Report

Add a button for Quark analysis at the nav menu.
Add Quark analysis report table that includes three columns:
1. **Potential malicious activity**: The description of potentially malicious activity detected by Quark.
2. **Source Code of Activity**: The class and method execute the APIs of potentially malicious activity.
3. **Confidence**: The number of confidence that the potentially malicious activity is practiced.

#### 2. Source code overview
Click on the link under the Source Code of Activity column, and it will open the exact .smali file and highlight the two APIs where the activities are executed.

Here is the demo:

### Checklist for PR

- [X] Run MobSF unit tests and lint `tox -e lint,test`
- [X] Tested Working on Linux, Mac, Windows, and Docker
- [X] Add unit test for any new Web API (Refer: `StaticAnalyzer/tests.py`)
- [X] Make sure tests are passing on your PR [MobSF tests](https://github.com/MobSF/Mobile-Security-Framework-MobSF/actions)

### CODE REVIEW

**Comment by ajinabraham** [comment]
This is great @pulorsok
Thanks for this PR 💯 
I will do a review and get back to you.

**Comment by superpoussin22** [comment]
@pulorsok 
I have a repeating error when analyzing apks:
`[ERROR] 15/Jul/2021 08:03:36 - Find line number of method and apis
Traceback (most recent call last):
  File "/Users/vince/MyDev/Mobile-Security-Framework-MobSF/mobsf/MalwareAnalyzer/views/quark.py", line 127, in _get_line_numbers
    if not method_found:
UnboundLocalError: local variable 'method_found' referenced before assignment`

File: mobsf/MalwareAnalyzer/views/quark.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        'second_api': -1,
+    }
+    try:
+        with open(source_code_path, 'r') as file:
+
+            for num, line in enumerate(file, 1):
+                if method in line and '.method' in line:
+                    method_found = True
+                    line_numbers['md_start'] = num
+
```
Comment (superpoussin22) [inline mobsf/MalwareAnalyzer/views/quark.py:126]: else:
    method_found = False

**Review by superpoussin22 — CHANGES_REQUESTED** [review]
see my proposal

**Review by ajinabraham — APPROVED** [review]
Looking good, did some QA. 
Quark is using `git` to update rules. Testing it on our docker image.

File: mobsf/MalwareAnalyzer/views/quark.py (hunk 1 above)
Comment (ajinabraham) [reply mobsf/MalwareAnalyzer/views/quark.py]: Addressed this.

**Review by superpoussin22 — APPROVED** [review]
LGTM 👍

**Pull request merged**